### PR TITLE
Update dexie.d.ts to version 1.2

### PR DESCRIPTION
**Pull request by prannayb**
From dexie.js source repo commit 92372b017810704bd7f812e7b0b86df866e8d222

### CODE REVIEW

File: dexie/dexie.d.ts [hunk 1]
```diff
@@ -306,7 +314,7 @@ declare module Dexie {
 
     interface IndexSpec {
         name: string;
-        keyPath: any;
+        keyPath: any; // string | Array<string>
```
Comment (horiuchi) [inline dexie/dexie.d.ts]: Why not change this type to `string | string[]`?

Comment (dfahlander) [inline dexie/dexie.d.ts]: It's just due to backward compatibility with old typescript compiler that don't accept multiple types. However, I believe it's soon time to start using the new typescript syntax but then there are several parts of the definition file that could be enhanced by changing from any to explicit types lists.

**Comment by dfahlander** [comment]
:+1:

**Comment by horiuchi** [comment]
@prannayb Thanks mate!
@dfahlander Thanks for review!

**Pull request merged**